src/objective/rank_xendcg_objective.hpp

```diff
@@ -0,0 +1,149 @@
+#ifndef LIGHTGBM_OBJECTIVE_RANK_XENDCG_OBJECTIVE_HPP_
+#define LIGHTGBM_OBJECTIVE_RANK_XENDCG_OBJECTIVE_HPP_
+
+#include <LightGBM/objective_function.h>
+#include <LightGBM/utils/common.h>
+#include <LightGBM/utils/random.h>
+
+#include <limits>
+#include <string>
+#include <algorithm>
+#include <cmath>
+#include <cstdio>
+#include <cstring>
+#include <math.h>
+#include <random>
+#include <vector>
+#include <random>
```

Comment:
I think includes can be refined after the latest refactoring. At least, there is a duplicated `<random>` entry.

Reply:
Thanks for catching this! I've cleaned up the includes.